src/emulator/controller.ts

```diff
@@ -320,6 +336,36 @@ export async function startAll(options: any): Promise<void> {
     await startEmulator(pubsubEmulator);
   }
 
+  if (shouldStart(options, Emulators.GUI) && EmulatorRegistry.isRunning(Emulators.HUB)) {
```

Comment:
nit: this is a little different to how we have done cross-emulator discovery so far.

In `firestoreEmulator` we do this in `start()`:

```
    const functionsPort = EmulatorRegistry.getPort(Emulators.FUNCTIONS);
    if (functionsPort) {
      this.args.functions_emulator = `localhost:${functionsPort}`;
    }
```

I would prefer `shouldStart()` be the only check here and you can throw an emulator inside `gui.start()` saying "The GUI cannot be started before the hub!" if you can't find the hub.

Reply:
Done. I think you meant `throw an exception` instead of `throw an emulator` though :).